File: AN_2/SEM_1/MAP/lab2/src/decorator/PrinterTaskRunner.java
package decorator;

import java.time.LocalDateTime;

public class PrinterTaskRunner extends AbstractTaskRunner
{

    public PrinterTaskRunner(TaskRunner taskRunner)
    {
        super(taskRunner);
    }

    @Override
    public void executeOneTask()
    {
        super.executeOneTask();
        System.out.println("Task: " + LocalDateTime.now());
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Decorator/PrinterTaskRunner.cs
using System;

namespace Decorator
{
    public class PrinterTaskRunner : AbstractTaskRunner
    {
        public PrinterTaskRunner(ITaskRunner taskRunner)
            : base(taskRunner)
        {
        }

        public override void ExecuteOneTask()
        {
            base.ExecuteOneTask();
            Console.WriteLine($"Task: {DateTime.Now}");
        }
    }
}
